### ISSUE TITLE
call_signatures improvements

### ISSUE THREAD

**Issue by florianlink** [davidhalter/jedi#512]
When inspecting the results of call_signatures, it would be useful to check if the contained call is a call to an instance or class method, to be able to show/hide the self argument. The current API does not support this, except by looking at the type of the internal _definition.

**Comment by davidhalter** [comment]
All right. Please recheck, I'm not sure what the issue is anymore. A few things like the call signatures for classes seem to be right now. Or that's at least what it looks like for me. You're invited to recheck.

This issue now is really about docstrings and how a call signature is represented in a docstring. But as I wrote before:

> Then again I still don't know what the docstring preferences are for class initializations in Python.

**Comment by florianlink** [comment]
Thanks, I will retry.

**Comment by davidhalter** [comment]
Can you summarize what this issue is about? I'm not sure.

**Comment by florianlink** [comment]
I did not yet have time to upgrade to a newer jedi version.
You can close this bug for now, I think.
